### PR TITLE
Clean architecture newER

### PR DESCRIPTION
**Pull request by BerkeAltiparmak**
Fixed Structure.
Fixed Naming.
Introduced new classes/files we need to have.
Got stuck in implementing some of those ideas; need to work together.
If seen appropriate, let's merge this and work on it.

### CODE REVIEW

**Comment by BerkeAltiparmak** [comment]
.save() should not be used within use cases. We need to be able to reach data processed by Use Case outside and only then we should save data.

File: server/src/UseCases/TranscriptUseCases/transcriptUploadInteractor.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    // save intents
+    try {
+        user.intents = allCurrentIntents
+        await user.save()
+    }
+    catch (e) {
+        throw new Error("Error in saving intents", e)
+    }
```
Comment (chenikabukes) [inline server/src/UseCases/TranscriptUseCases/transcriptUploadInteractor.js:37]: Great job adding error catching when saving the transcript. 
One thing I've encountered is when I upload two transcripts of the same name then an _Axios Error: Network Error_ is displayed, try it yourself and then see if you can catch this error too.

**Review by chenikabukes — CHANGES_REQUESTED** [review]

File: server/src/UseCases/TranscriptUseCases/transcriptUploadInteractor.js (hunk 1 above)
Comment (BerkeAltiparmak) [reply server/src/UseCases/TranscriptUseCases/transcriptUploadInteractor.js:37]: yes I've noticed the same error as well, good catch! Let me fix this by the end of the day.

**Comment by chenikabukes** [comment]
Nice that the error no longer appears. 
With regards to the new bump in the road, I will work on the front end side of that error within the coming days. 
For future reference, I recommend keeping commit messages shorter and rather commenting on the pull request about what you changed so that the commits remain straightforward.
Great job on this progress!

**Comment by aidanmrli** [comment]
Comments after Thurs 17 Nov TLI session: we do not necessarily need input/output boundary as the endpoints in the controllers are literally the input/output boundaries. So it is perfectly fine for the Controller to call the Use Case Interactor directly instead of an input/output data boundary as long as the Use Case Interactor does not know about the existence of a Controller.

We should add comments to the entities schema file - technically, because we are so reliant on MongoDB and the mongoose schema wrapper, this isn't clean. However, because this is a MERN app, it is focused on MongoDB. Ideally, we would create an abstract class or something for an entity and implement the class in Mongo in a data persistence layer, possibly not even using Mongoose schema. However, this is too much work and it's not feasible given the scope of our project (need to alter every file in the backend).

**Comment by aidanmrli** [comment]
Need to implement testing for use cases.

**Pull request merged**